scipy/optimize/_linprog_util.py

```diff
@@ -865,25 +950,18 @@ def _parse_linprog(c, A_ub, b_ub, A_eq, b_eq, bounds, options, x0):
                 Set to True to print convergence messages.
 
         For method-specific options, see :func:`show_options('linprog')`.
-    x0 : 1D array, optional
-        Starting values of the independent variables, which will be refined by
-        the optimization algorithm. Currently compatible only with the
-        'revised simplex' method, and only if x0 is a basic feasible solution
-        of the problem.
     """
     if options is None:
         options = {}
 
     solver_options = {k: v for k, v in options.items()}
-    solver_options, A_ub, A_eq = _check_sparse_inputs(solver_options, A_ub, A_eq)
+    solver_options, A_ub, A_eq = _check_sparse_inputs(solver_options, lp.A_ub, lp.A_eq)
     # Convert lists to numpy arrays, etc...
-    c, A_ub, b_ub, A_eq, b_eq, bounds, x0 = _clean_inputs(
-        c, A_ub, b_ub, A_eq, b_eq, bounds, x0)
-    return c, A_ub, b_ub, A_eq, b_eq, bounds, solver_options, x0
+    lp = _clean_inputs(lp._replace(A_ub=A_ub, A_eq=A_eq))
+    return lp, solver_options
```

Comment:
Perhaps it would be better to pass ``lp`` to ``_check_sparse_inputs`` directly? That way we won't have to think about explicitly replacing `A_ub`/`A_eq`. So  it would become:
```python
lp, solver_options = _check_sparse_inputs(solver_options, lp)
lp = _clean_inputs(lp)
return lp, solver_options
```
Looking at it now, if we change it, `lp` would need to be more descriptive

Reply:
@Kai-Striega Is this resolved?